plugins/CustomDimensions/vue/src/Edit/Edit.vue

```diff
@@ -0,0 +1,350 @@
+      this.init();
+    },
+  },
+  methods: {
+    removeAnyCustomDimensionNotification() {
+      NotificationsStore.remove(notificationId);
+    },
+    showNotification(message: string, context: NotificationType['context']) {
+      NotificationsStore.show({
+        message,
+        context,
+        id: notificationId,
+        type: 'transient',
+      });
+    },
+    init() {
+      if (this.dimensionId !== null) {
+        this.removeAnyCustomDimensionNotification();
+      }
+
+      CustomDimensionsStore.fetch().then(() => {
+        if (this.edit && this.dimensionId) {
+          this.dimension = clone(
+            CustomDimensionsStore.customDimensionsById.value[this.dimensionId],
+          ) as unknown as CustomDimension;
+
+          if (this.dimension && !this.dimension.extractions.length) {
+            this.addExtraction();
+          }
+        } else if (this.create) {
+          this.dimension = {
+            idsite: Matomo.idSite,
+            name: '',
+            active: false,
+            extractions: [],
+            scope: this.dimensionScope,
+            case_sensitive: true,
+          } as unknown as CustomDimension;
+          this.addExtraction();
+        }
+      });
+    },
+    removeExtraction(index: number) {
+      if (index > -1) {
+        this.dimension.extractions.splice(index, 1);
+      }
+    },
+    addExtraction() {
+      if (this.doesScopeSupportExtraction) {
+        this.dimension.extractions.push({
+          dimension: 'url',
+          pattern: '',
+        });
+      }
+    },
+    createCustomDimension() {
+      this.isUpdatingDim = true;
+      CustomDimensionsStore.createOrUpdateDimension(
+        this.dimension,
+        'CustomDimensions.configureNewCustomDimension',
+      ).then(() => {
+        this.showNotification(translate('CustomDimensions_DimensionCreated'), 'success');
+        console.log('before reload');
+        CustomDimensionsStore.reload();
+        MatomoUrl.updateHashToUrl('/list');
+      }).finally(() => {
+        this.isUpdatingDim = false;
+      });
+    },
+    updateCustomDimension() {
+      this.isUpdatingDim = true;
+      CustomDimensionsStore.createOrUpdateDimension(
+        this.dimension,
+        'CustomDimensions.configureExistingCustomDimension',
+      ).then(() => {
+        this.showNotification(translate('CustomDimensions_DimensionUpdated'), 'success');
+        CustomDimensionsStore.reload();
+        MatomoUrl.updateHashToUrl('/list');
+      }).finally(() => {
+        this.isUpdatingDim = false;
+      });
+    },
+    manuallyTrackCodeViaJs(dimension: CustomDimension) {
+      return `_paq.push(['setCustomDimension', ${dimension.idcustomdimension}, `
+        + `'${translate('CustomDimensions_ExampleValue')}']);`;
+    },
+    manuallyTrackCodeViaPhp(dimension: CustomDimension) {
+      return `$tracker->setCustomDimension('${dimension.idcustomdimension}', `
+        + `'${translate('CustomDimensions_ExampleValue')}');`;
+    },
```

Comment:
Shouldn't those be computed properties like `manuallyTrackCode`? Currently it only outputs 
![image](https://user-images.githubusercontent.com/1579355/158790466-761bec64-ecc9-4480-86ea-781da2bff5f9.png)

Reply:
yes, fixed, apologies for the back & forth

Reply:
actually they should be methods, not yet fixed

Reply:
should be fixed in latest commit